### PR TITLE
extract kubermatic CRD's out of the kubermatic helm chart

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Extract kubermatic CRD's out of the kubermatic helm chart

**Documentation**:
https://github.com/kubermatic/docs/pull/84

**Release note**:
```release-note
[ACTION_REQUIRED] Kubermatic CustomResourceDefinitions have been extracted out of the helm chart. This requires the execution of the `charts/kubermatic/migrate/migrate-kubermatic-chart.sh` script in case the CRD's where installed without the `"helm.sh/resource-policy": keep` annotation.
```

### CODE REVIEW

**Comment by mrIncompetent** [comment]
Docs PR is missing

**Comment by mrIncompetent** [comment]
Docs PR opened: https://github.com/kubermatic/docs/pull/84

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

File: config/kubermatic/migrate/migrate-kubermatic-chart.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
```
Comment (alvaroaleman) [inline config/kubermatic/migrate/migrate-kubermatic-chart.sh:1]: This script absolutely needs a `set -euo pipefail`

File: config/kubermatic/migrate/migrate-kubermatic-chart.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+kubectl delete clusterrolebindings.rbac.authorization.k8s.io kubermatic
+
+for cm in $(kubectl -n kubermatic-installer get configmap -o json | jq -r '.items[].metadata.name' | grep 'kubermatic');do
+    echo "Deleting helm release info in ConfigMap: ${TILLER_NAMESPACE}/${cm}"
+    kubectl -n ${TILLER_NAMESPACE} delete configmap ${cm}
+done
+
+helm upgrade --install --tiller-namespace=${TILLER_NAMESPACE} \
+    --values ${VALUES_FILE} \
+    --namespace kubermatic kubermatic ./kubermatic/ --dry-run --debug
```
Comment (alvaroaleman) [inline config/kubermatic/migrate/migrate-kubermatic-chart.sh]: You probably do not want this to be a `--dry-run`?

File: config/kubermatic/migrate/migrate-kubermatic-chart.sh (hunk 1 above)
Comment (mrIncompetent) [reply config/kubermatic/migrate/migrate-kubermatic-chart.sh:1]: done in https://github.com/kubermatic/kubermatic/pull/2459/commits/31e500e58b639e0f8b2c58ce10ad1103d831b836

File: config/kubermatic/migrate/migrate-kubermatic-chart.sh (hunk 2 above)
Comment (mrIncompetent) [reply config/kubermatic/migrate/migrate-kubermatic-chart.sh]: removed with https://github.com/kubermatic/kubermatic/pull/2459/commits/9b07c3449a84c25b696aca21801444647fdf69ea

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**